app/Main.hs

```diff
@@ -99,13 +123,166 @@ makePscPackage force = do
+    hasSpacchettiDhall <- T.testfile spacchettiDhallPath
+    T.when hasSpacchettiDhall $ T.die
+       $ "Found " <> unsafePathToText spacchettiDhallPath <> ": there's already a project here. "
+      <> "Run `spacchetti initSpacchetti --force` if you're sure you want to overwrite it."
+  T.touch spacchettiDhallPath
+  T.writeTextFile spacchettiDhallPath Templates.spacchettiDhall
+  T.void $ T.shell ("dhall format --inplace " <> spacchettiDhallText) T.empty
+  where
+    spacchettiDhallText = "spacchetti.dhall"
+    spacchettiDhallPath = T.fromText spacchettiDhallText
+
 localSetup :: Bool -> IO ()
 localSetup force = do
-  makeDhall force
+  makePackagesDhall force
   makePscPackage force
   T.echo "Set up local Spacchetti packages."
   T.echo "Run `spacchetti insdhall` to generate the package set."
 
+initSpacchetti :: Bool -> IO ()
+initSpacchetti force = do
+  makePackagesDhall force
+  makeSpacchetti force
+  T.mktree "src"
+  T.mktree "test"
+  T.writeTextFile "src/Main.purs" Templates.srcMain
+  T.writeTextFile "test/Main.purs" Templates.testMain
+  T.writeTextFile ".gitignore" Templates.gitignore
+  T.echo "Set up a local Spacchetti project."
+  T.echo "Try running `spacchetti install`"
+
+spacchettiJsonPath :: Text.Text
+spacchettiJsonPath = spacchettiDir <> "spacchetti.json"
+
+dhallToJsonCmd :: Text.Text
+dhallToJsonCmd = "cat ./spacchetti.dhall | dhall-to-json --pretty > " <> spacchettiJsonPath
+
+makeSpacchettiJson :: IO ()
+makeSpacchettiJson = do
+  T.mktree $ T.fromText spacchettiDir
+  code <- T.shell dhallToJsonCmd T.empty
+  case code of
+    T.ExitFailure n -> T.die ("making spacchetti.json failed: " <> T.repr n)
+    T.ExitSuccess -> pure ()
+
+readSpacchettiJson :: IO Types.SpacchettiConfig
+readSpacchettiJson = do
+  spacchettiConfigText <- T.readTextFile (T.fromText spacchettiJsonPath)
+  case eitherDecodeStrict $ Text.encodeUtf8 spacchettiConfigText of
+    Left e -> T.die $ "The generated spacchetti.json was in the wrong format: " <> Text.pack e
+    Right config -> return config
+
+install :: IO ()
+install = do
+  T.echo "generating spacchetti.json"
+  makeSpacchettiJson
+  spacchettiConfigText <- T.readTextFile (T.fromText spacchettiJsonPath)
+  config <- readSpacchettiJson
+  let deps = getAllDependencies config
+  echo' $ "installing " <> Text.pack (show $ List.length deps) <> " dependencies."
+  traverse getDep deps
+  T.echo "installation complete."
+
+ensureSpacchettiJson :: IO ()
+ensureSpacchettiJson = do
+  exists <- T.testfile $ T.fromText spacchettiJsonPath
+  T.unless exists makeSpacchettiJson
+
+getDir :: (Types.PackageName, Types.PackageDefinition) -> Text.Text
+getDir
+  ( Types.PackageName name
+  , Types.PackageDefinition
+      { Types.version=version
+      , Types.repo=repo
+      }
+  )
+  = spacchettiDir <> name <> "/" <> version
+
+getGlobs :: [(Types.PackageName, Types.PackageDefinition)] -> [Text.Text]
+getGlobs = map fn
+  where
+    fn pair = (getDir pair) <> "/src/**/*.purs"
```

Comment:
Do we want to hardcode the globs? I think it would make sense to make them configurable in the `spacchetti.dhall`

Reply:
Mmmmmmmmmm, maybe that's a good place to put that